scripts/install/install-dependencies.sh

```diff
@@ -0,0 +1,394 @@
+  exit 1
+fi
+
+cd "${PANDIR}"
+
+# TODO(jamessynge): Add flags to control behavior, such as skipping apt-get.
+
+ASTROMETRY_VERSION="0.72"
+INSTALL_PREFIX="/usr/local"
+DO_APT_GET=1
+DO_MONGODB=1
+DO_CONDA=1
+DO_CFITSIO=0
+DO_ASTROMETRY=1
+DO_ASTROMETRY_INDICES=1
+DO_ASTROMETRY_FULL_INDICES=0
+DO_PIP_REQUIREMENTS=1
+
+function echo_bar() {
+  eval $(resize|grep COLUMNS=)
+  printf "%${COLUMNS:-80}s\n" | tr ' ' '#'
+}
+
+# Append $1 to PATH and write command to do the same to .bashrc.
+function add_to_PATH() {
+  local -r the_dir="$(readlink -f "${1}")"
+  if [[ -z "$(egrep -- "PATH=.*${the_dir}" ~/.bashrc)" ]] ; then
+    echo >>"${HOME}/.bashrc" "
+# Added by PANOPTES install-dependencies.sh
+PATH=\"${the_dir}:\${PATH}\""
+    PATH="${the_dir}:${PATH}"
+    echo
+    echo "Added ${the_dir} to PATH in .bashrc"
+  else
+    echo ".bashrc already adds ${the_dir} to PATH"
+  fi
+}
+
+# Given the path to a pkg-config file (.pc), extract the version number.
+function extract_version_from_pkg_config() {
+  if [[ -f "${1}" ]] ; then
+    egrep '^Version:' "${1}" | cut '-d:' -f2
+  else
+    echo ""
+  fi
+}
+
+function install_apt_packages() {
+  # Remove all the comments from the package list and install the packages whose
+  # names are left.
+  APT_PKGS="$(cut '-d#' -f1 "${THIS_DIR}/apt-packages-list.txt" | sort | uniq)"
+  echo
+  echo "Running sudo apt-get install, you may be prompted for your password."
+  echo
+  (set -x ; sudo apt-get update)
+  (set -x ; sudo apt-get install --yes ${APT_PKGS})
+}
+
+function install_mongodb() {
+  # This is based on https://www.howtoforge.com/tutorial/install-mongodb-on-ubuntu-16.04/
+  # Note this function does not configure mongodb itself, i.e. no users or
+  # security settings.
+  echo_bar
+  echo "
+Installing mongodb, for which several commands require sudo, so
+you may be prompted for you password.
+"
+  sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv EA312927
+  echo "deb http://repo.mongodb.org/apt/ubuntu "$(lsb_release -sc)"/mongodb-org/3.2 multiverse" \
+     | sudo tee /etc/apt/sources.list.d/mongodb-org-3.2.list
+  sudo apt-get update
+  sudo apt-get install -y mongodb-org
+  echo "[Unit]
+Description=High-performance, schema-free document-oriented database
+After=network.target
+Documentation=https://docs.mongodb.org/manual
+
+[Service]
+User=mongodb
+Group=mongodb
+ExecStart=/usr/bin/mongod --quiet --config /etc/mongod.conf
+
+[Install]
+WantedBy=multi-user.target" | sudo tee /lib/systemd/system/mongod.service
+
+  sudo systemctl daemon-reload
+  sudo systemctl start mongod
+  sudo systemctl enable mongod
+}
+
+function maybe_install_mongodb() {
+  if [[ -z "$(which mongo)" || -z "$(systemctl | grep mongo)" ]] ; then
+    install_mongodb
+  fi
+}
+
+# Install miniconda (conda with no additional packages); we can then select
+# the set to install.
+function install_conda() {
+  local -r the_script="${PANDIR}/tmp/miniconda.sh"
+  echo_bar
+  echo
+  echo "Installing miniconda"
```

Comment:
Just to be good sports we should mention here that there is a default acceptance of the license agreement.

Reply:
Added link to license.